solidity/src/contracts/PAM.sol

```diff
@@ -150,9 +150,11 @@ contract PAM is Ownable, IPAM {
         uint256 amount = uint256(bytes32(content[offset:offset += 32]));
         bytes32 sender = bytes32(content[offset:offset += 32]);
         uint256 recipientLen = uint256(bytes32(content[offset:offset += 32]));
-        address recipient = _bytesToAddress(
-            content[offset:offset += recipientLen]
-        );
+        bytes memory _recipient = recipientLen == 40 ?
+            _addHexPrefixToBytes(content[offset:offset += recipientLen]) :
+            content[offset:offset += recipientLen];
+        address recipient = _bytesToAddress(_recipient);
+        
```

Comment:
Uhm looks expensive, are there no other ways? Like checking the length within `_bytesToAddress` and if it's 42 then skip the first two bytes otherwise don't

Reply:
I didn't check `_bytesToAddress` and went for the expensive road.
fixed